openedx/core/djangoapps/user_api/api/profile.py

```diff
@@ -90,14 +90,14 @@ def update_preferences(username, **kwargs):
 
 
 @intercept_errors(ProfileInternalError, ignore_errors=[ProfileRequestError])
-def update_email_opt_in(username, org, optin):
+def update_email_opt_in(user, org, optin):
```

Comment:
Note that one of the places where this was called (UpdateEmailOptInPreference in /edx-platform/openedx/core/djangoapps/user_api/views.py) was already passing in the user instead of the username. It appears that the code didn't break because unicode(user) == username.